src/api/schema/filter.rs

```diff
@@ -46,8 +47,38 @@ impl StringFilter {
 }
 
 #[derive(InputObject)]
-#[graphql(concrete(name = "ComponentKindFilter", params(ComponentKind)))]
 #[graphql(concrete(name = "SourceOutputTypeFilter", params(source::SourceOutputType)))]
+// Filter for GraphQL lists
+pub struct ListFilter<T: InputType + PartialEq + Eq + Ord> {
+    pub equals: Option<Vec<T>>,
+    pub not_equals: Option<Vec<T>>,
+    pub contains: Option<T>,
+    pub not_contains: Option<T>,
+}
+
+impl<T: InputType + PartialEq + Eq + std::cmp::Ord> ListFilter<T> {
+    pub fn filter_value(&self, value: Vec<T>) -> bool {
+        let val = BTreeSet::from_iter(value.iter());
+        filter_check!(
+            // Equals
+            self.equals
+                .as_ref()
+                .map(|s| BTreeSet::from_iter(s.iter()).eq(&val)),
+            // Not Equals
+            self.not_equals
+                .as_ref()
+                .map(|s| !BTreeSet::from_iter(s.iter()).eq(&val)),
+            // Contains
+            self.contains.as_ref().map(|s| val.contains(s)),
+            // Not Contains
+            self.not_contains.as_ref().map(|s| !val.contains(s))
+        );
+        true
+    }
+}
+
+#[derive(InputObject)]
+#[graphql(concrete(name = "ComponentKindFilter", params(ComponentKind)))]
```

Comment:
Nit: Since you introduced the `source::SourceOutputType` pattern, you could change this param to `source::ComponentKind` and drop the explicit import.

I like this better- it's clearer that it's being implemented on source components rather than having to check the import list for the scope.